opensrp-goldsmith/src/main/java/org/smartregister/goldsmith/interactor/DefaultAncHomeVisitInteractorFlv.java

```diff
@@ -0,0 +1,282 @@
+                .withOptional(false)
+                .withDetails(details)
+                .withHelper(new ANCCounselingAction(memberObject, dateMap))
+                .withFormName(Constants.JSON_FORM.ANC_HOME_VISIT.getAncCounseling())
+                .build();
+        actionList.put(context.getString(R.string.anc_home_visit_counseling), counseling);
+    }
+
+    protected void evaluateSleepingUnderLLITN() throws Exception {
+        BaseAncHomeVisitAction sleeping = new BaseAncHomeVisitAction.Builder(context, context.getString(R.string.anc_home_visit_sleeping_under_llitn_net))
+                .withOptional(false)
+                .withDetails(details)
+                .withHelper(new SleepingUnderLLITNAction())
+                .withDestinationFragment(BaseAncHomeVisitFragment.getInstance(view, Constants.JSON_FORM.ANC_HOME_VISIT.getSleepingUnderLlitn(), null, details, null))
+                .build();
+
+        actionList.put(context.getString(R.string.anc_home_visit_sleeping_under_llitn_net), sleeping);
+    }
+
+    protected void evaluateANCCard() throws Exception {
+        if (memberObject.getHasAncCard() != null && memberObject.getHasAncCard().equals("Yes")) {
+            return;
+        }
+
+        BaseAncHomeVisitAction anc_card = new BaseAncHomeVisitAction.Builder(context, context.getString(R.string.anc_home_visit_anc_card_received))
+                .withOptional(false)
+                .withDetails(details)
+                .withHelper(new ANCCardAction())
+                .withDestinationFragment(BaseAncHomeVisitFragment.getInstance(view, Constants.JSON_FORM.ANC_HOME_VISIT.getAncCardReceived(), null, details, null))
+                .build();
+
+        actionList.put(context.getString(R.string.anc_home_visit_anc_card_received), anc_card);
+    }
+
+    protected void evaluateHealthFacilityVisit(Map<Integer, LocalDate> dateMap) throws Exception {
+        String visit_title = MessageFormat.format(context.getString(R.string.anc_home_visit_facility_visit), memberObject.getConfirmedContacts() + 1);
+        BaseAncHomeVisitAction facility_visit = new BaseAncHomeVisitAction.Builder(context, visit_title)
+                .withOptional(false)
+                .withDetails(details)
+                .withHelper(new HealthFacilityVisitAction(memberObject, dateMap))
+                .withFormName(Constants.JSON_FORM.ANC_HOME_VISIT.getHealthFacilityVisit())
+                .build();
+
+        actionList.put(visit_title, facility_visit);
+    }
+
+    protected void evaluateTTImmunization(VaccineTaskModel vaccineTaskModel) throws Exception {
+        if (ifPendingVaccines(vaccineTaskModel))
+            return;
+
+        // compute the due date
+        final Triple<DateTime, VaccineRepo.Vaccine, String> individualVaccine = VaccineScheduleUtil.getIndividualVaccine(vaccineTaskModel, "TT");// if there are no pending vaccines
+
+        if (vaccineOverdue(individualVaccine))
```

Comment:
the method naming is misleading. I think this checks if the vaccine is not overdue i.e. the date is after today & not before

Reply:
This is now fixed